### ISSUE TITLE
False positive: UnusedVariable for methods that unconditionally throw

### ISSUE THREAD

**Issue by electrum** [google/error-prone#1948]
### Description of the problem / feature request:

*UnusedVariable* triggers for a method like this:

```java
private static boolean compare(int left, int right)
{
    throw new AssertionError("this method should not be called");
}
```

We can rename the parameters to `unusedLeft` and `unusedRight`, which is often a good way to signal intent to readers of the code, but that is redundant in this case, since it's immediately obvious that this method will not use the parameters.

### What version of Error Prone are you using?

2.4.0

**Comment by cushon** [comment]
Note that this check only reports unused parameters on private methods, and it doesn't seem that often that a private method that unconditionally throws would be useful? Do you have more context about the situation where you encountered this problem?

**Comment by electrum** [comment]
I admit this situation is quite specialized. Here's the specific example: https://github.com/trinodb/trino/blob/351/core/trino-main/src/main/java/io/trino/type/UnknownType.java#L138

This class the implementation of a type in a SQL engine. The method, annotated with `@ScalarOperator`, will be discovered and invoked via generated code. It's private because it's only for use by the engine and shouldn't be called directly -- it's not a public API for general consumption. The unused arguments are present because they are part of the expected "API" that the annotated method needs to implement.
